website/static/js/registrationEditorExtensions.js

```diff
@@ -169,45 +167,73 @@ var Uploader = function(question) {
     question.showUploader = ko.observable(false);
     question.uid = 'uploader_' + uploaderCount;
     uploaderCount++;
-    self.selectedFile = ko.observable({});
-    self.selectedFile.subscribe(function(file) {
-        if (file) {
-            question.extra({
-                selectedFileName: file.data.name,
-                nodeId: file.data.nodeId,
-                viewUrl: '/project/' + file.data.nodeId + '/files/osfstorage' + file.data.path,
-                sha256: file.data.extra.hashes.sha256,
-                hasSelectedFile: true
-            });
-            question.value(file.data.name);
-        }
-        else {
-            question.extra({
-                selectedFileName: NO_FILE
-            });
-            question.value(NO_FILE);
-        }
+    self.selectedFiles = ko.observableArray(question.extra() || []);
+    self.selectedFiles.subscribe(function(fileList) {
+        $.each(fileList, function(idx, file) {
+            if (file && !self.fileAlreadySelected(file)) {
+                question.extra().push(file);
+            }
+        });
+        question.value(question.formattedFileList());
     });
+
+    self.addFile = function(file) {
+        if(self.fileAlreadySelected(file))
+            return false;
+
+        self.selectedFiles.push({
+            data: file.data,
+            selectedFileName: file.data.name,
+            nodeId: file.data.nodeId,
+            viewUrl: '/project/' + file.data.nodeId + '/files/osfstorage' + file.data.path,
+            sha256: file.data.extra.hashes.sha256
+        });
+        return true;
+    };
+
+
+    self.fileAlreadySelected = function(file) {
+        var selected = false;
+        $.each(question.extra(), function(idx, alreadyFile) {
+            if(alreadyFile.sha256 === file.data.extra.hashes.sha256){
+                selected = true;
+                return;
+            }
+        });
+        return selected;
+    };
+
     self.hasSelectedFile = ko.computed(function() {
-        return !!(question.extra().viewUrl);
+        return question.extra().length !== 0;
     });
-    self.unselectFile = function() {
-        self.selectedFile(null);
-        question.extra({
-            selectedFileName: NO_FILE
-        });
+    self.unselectFile = function(fileToRemove) {
+
+        var files = question.extra();
+
+        for (var i = 0; i < files.length - 1; i++) {
```

Comment:
This prevents a user from removing the last selected file. `i < files.length` or `i <= files.length - 1` would fix.